tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -1,9 +1,302 @@
+    await mito.getByRole('button', { name: 'Import test.csv' }).click();
+    await mito.getByRole('button', { name: 'Import test.csv' }).click();
+    await mito.locator('#mito-center-content-container line').nth(1).click();
+    await mito.getByText('Column3').click();
+    await mito.locator('[id="mito-toolbar-button-add\\ column\\ to\\ the\\ right"]').getByRole('button', { name: 'Insert' }).click();
+    await mito.locator('#cell-editor-input').fill('=123');
+    await mito.locator('#cell-editor-input').press('Enter');
+  
+  });
+
+  test.skip('Color Dataframe', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Format', exact: true })
+
+    await checkOpenTaskpane(mito, 'Color Dataframe');
+
+    await mito.getByText('Column Headers').click();
+    await mito.getByRole('textbox').first().fill('#3850b7');
+
+    // Expect the first column header to have a background color of background-color: rgb(56, 80, 183)
+    await expect(mito.locator('.endo-column-header-container').first()).toHaveAttribute('style', /background-color: rgb\(56, 80, 183\)/);
+  });
+
+  test('Delete Column', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await mito.getByTitle('Column1').click();
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Delete' })
+
+    await expect(mito.getByText('Column1')).not.toBeVisible();
+  });
+
+  test('Insert Column', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await mito.getByTitle('Column1').click();
+    await mito.locator('[id="mito-toolbar-button-add\\ column\\ to\\ the\\ right"]').getByRole('button', { name: 'Insert' }).click();
+    await awaitResponse(page);
+
+    // Expect there to be 4 column headers
+    await expect(mito.locator('.endo-column-header-container')).toHaveCount(4);
+  });
+
+  test('Filter', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Filter' })
+
+    await checkOpenTaskpane(mito, 'Column1');
+
+    // Filter out all rows, with a > 4 filter
+    await mito.getByText('+ Add Filter').click();
+    await mito.locator('div').filter({ hasText: /^Add a Filter$/ }).first().click();
+    await mito.getByRole('textbox').click();
+    await mito.getByRole('textbox').fill('4');
+    await mito.getByText('No rows in dataframe.').click();
+
+    // Add another filter and combine with an OR < 10
+    await mito.getByText('+ Add Filter').click();
+    await mito.locator('div').filter({ hasText: /^Add a Filter$/ }).first().click();
+    await mito.getByText('And').click();
+    await mito.getByText('Or', { exact: true }).click();
+    await mito.getByText('>').nth(1).click();
+    await mito.getByText('<').click();
+    await mito.locator('div').filter({ hasText: /^Or<$/ }).getByRole('textbox').fill('10');
```

Comment:
Want to test that the `result` text is updated correctly? The text that says `Removed an additional XXX rows`